bridge/slack/slack.go

```diff
@@ -315,6 +321,39 @@ func (b *Bslack) sendRTM(msg config.Message) (string, error) {
 	return b.postMessage(&msg, messageParameters, channelInfo)
 }
 
+func (b *Bslack) updateTopicOrPurpose(msg *config.Message, channelInfo *slack.Channel) (handled bool, err error) {
+	if msg.Event != config.EVENT_TOPIC_CHANGE {
+		return false, nil
+	}
+
+	if !b.GetBool("SyncTopicChange") {
+		return false, nil
+	}
+
+	incomingChangeType, text := b.extractTopicOrPurpose(msg.Text)
+	switch incomingChangeType {
+	case "topic":
+		if strings.HasSuffix(channelInfo.Topic.Value, "[nosync]") {
+			break
+		}
+		_, err = b.rtm.SetTopicOfConversation(channelInfo.ID, text)
```

Comment:
Errors should be handled at this point directly:
```suggestion
		for {
			_, err = b.rtm.SetTopicOfConversation(channelInfo.ID, text)
			if err == nil {
				return true, nil
			}
			if err = b.handleRateLimit(err); err != nil {
				return true, err
			}
		}
```